qt/traffic_mode.hpp

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+#include "indexer/feature.hpp"
+
+#include "openlr/openlr_model.hpp"
+#include "openlr/openlr_sample.hpp"
+
+#include "std/string.hpp"
+#include "std/unique_ptr.hpp"
+#include "std/unordered_map.hpp"
+
+#include <QAbstractTableModel>
+
+class Index;
+class QItemSelection;
+
+struct DecodedSample
+{
+  DecodedSample(Index const & index, openlr::SamplePool const & sample);
+
+  openlr::SamplePool const & GetItems() const { return m_decodedItems; }
+  vector<m2::PointD> GetPoints(size_t const index) const;
+
+  map<FeatureID, FeatureType> m_features;
```

Comment:
Why not to directly store feature points, instead of whole feature? It's quite unsafe to store FeatureType in a map as you need to guarantee that ParseEverything() is called.